### ISSUE TITLE
I don't understand the viewer\editor

### ISSUE THREAD

**Issue by opcodewriter** [sqlitebrowser/sqlitebrowser#1416]
I'm looking at a column in a row which is of INTEGER type
![snip_20180607095319](https://user-images.githubusercontent.com/15019084/41082976-875bfd56-6a1f-11e8-9b26-bc0f2cb4a182.png)

I am having a hard time understanding the left side marked in red. Isn't this supposed to show the raw binary data?

It looks to me that like it's transforming the integer value into a string and then it's showing the hexadecimal representation of the string.

Instead of "31", I was expecting to see "00 00 00 01" which is the 4 byte hexa representation of a 32 bits integer. This is how hex viewers would show the value:

![snip_20180607100049](https://user-images.githubusercontent.com/15019084/41083393-a378ad62-6a20-11e8-9fe5-977fb4590123.png)

Can someone explain how the viewer\editor works in SQLiteBrowser?

Is there a way to see the raw binary data in SQLteBrowser ?

**Comment by chrisjlocke** [comment]
Yes, you're seeing the hex value of "1", not the hex value of each byte.

**Comment by opcodewriter** [comment]
@chrisjlocke Thanks. But isn't that misleading? If you want to see exact raw representation, this can be confusing

**Comment by justinclift** [comment]
Oh yeah.  That's a good point.  Since the column is defined as an integer at the table level, it could be unexpected to see the string representation of it being shown in the hex editor view.

It might not be that difficult to change.  We'd have to have an awareness of the declared field type (as per the table creation statement), and an awareness of the value actually stored in the specific field being shown.  If it's an actual string value being stored there (due to SQLite flexible data typing) then the current behaviour would be ok.  But for actual integer values, we'd probably want to display thing how you're expecting them.  eg `0x01` for `1` (and so on) :smile:

**Comment by mgrojo** [comment]
We don't handle correctly the case for integer data. In fact the label: "Type of data currently in cell: Text / Numeric" is already giving a hint. We only treat blobs as binary data. Numeric data is treated as text, even the rowid column is behaving as text. That is not ideal but I wonder that the implementation focused on blob data.

**Comment by sky5walk** [comment]
Wow, I learned something new. 😨 
Ultimately, DB4S can never alter the original data of any type just by selecting a different view in the cell editor. Instead, drop auto-detect of core data in the cell editor and add more viewer types that intentionally interpret the core data.
Examples:
- Binary viewer = no conversion, just bytes.
- XML, SQL, JSON, etc viewer = attempt to display core data with those text parsers.

**Comment by polarbearing** [comment]
> "Type of data currently in cell: Text / Numeric" is already giving a hint.

No it isn't. I have a column here that is defined as VARCHAR and has the string "ok" in it, and is also shown as "Type of data currently in cell: Text / Numeric".
<img width="589" src="https://user-images.githubusercontent.com/8266355/106382298-e44b2c80-63be-11eb-8bcc-30b47e0b4745.png">

**Comment by chrisjlocke** [comment]
Reading back skywalks comments and then yours, I think there is a bit of miscommunication going on.
Skywalk is right - as per your screenshot - the value of 'OK' is either 'text/numeric' which is what DB4S is showing.  It's not a blob or an image...
Varchar isn't a valid sqlite type anyway, so unsure why you'd have a column defined as that.

**Comment by mgrojo** [comment]
> > "Type of data currently in cell: Text / Numeric" is already giving a hint.
> 
> No it isn't. I have a column here that is defined as VARCHAR and has the string "ok" in it, and is also shown as "Type of data currently in cell: Text / Numeric".

I meant, that it's already giving a hint about our behaviour of not distinguishing between numeric or text data.
